concent_api/concent_api/tests/test_shutdown_mode.py

```diff
@@ -0,0 +1,329 @@
+            config.SHUTDOWN_MODE = False
+            store_subtask(
+                task_id                 = '1',
+                subtask_id              = '8',
+                provider_public_key     = self.PROVIDER_PUBLIC_KEY,
+                requestor_public_key    = self.REQUESTOR_PUBLIC_KEY,
+                state                   = Subtask.SubtaskState.FORCING_REPORT,
+                next_deadline           = get_current_utc_timestamp() + settings.CONCENT_MESSAGING_TIME,
+                task_to_compute         = task_to_compute,
+                report_computed_task    = report_computed_task,
+            )
+            config.SHUTDOWN_MODE = True
+
+        self.stored_message_counter = 2
+
+        serialized_ack_report_computed_task = self._get_serialized_ack_report_computed_task(
+            timestamp                = "2017-12-01 11:00:05",
+            ack_report_computed_task = self._get_deserialized_ack_report_computed_task(
+                timestamp       = "2017-12-01 11:00:05",
+                subtask_id      = '8',
+                task_to_compute = task_to_compute
+            ),
+            requestor_private_key    = self.REQUESTOR_PRIVATE_KEY
+        )
+
+        with freeze_time("2017-12-01 11:00:05"):
+            response = self.client.post(
+                reverse('core:send'),
+                data                           = serialized_ack_report_computed_task,
+                content_type                   = 'application/octet-stream',
+                HTTP_CONCENT_CLIENT_PUBLIC_KEY = self._get_encoded_requestor_public_key(),
+            )
+
+        self.assertEqual(response.status_code,  202)
+        self.assertEqual(len(response.content), 0)
+        self._assert_stored_message_counter_increased(increased_by = 1)
+        self._test_subtask_state(
+            task_id                  = '1',
+            subtask_id               = '8',
+            subtask_state            = Subtask.SubtaskState.REPORTED,
+            provider_key             = self._get_encoded_provider_public_key(),
+            requestor_key            = self._get_encoded_requestor_public_key(),
+            expected_nested_messages = {'task_to_compute', 'report_computed_task', 'ack_report_computed_task'},
+        )
+        self._test_last_stored_messages(
+            expected_messages= [
+                message.concents.AckReportComputedTask,
+            ],
+            task_id         = '1',
+            subtask_id      = '8',
+            timestamp       = "2017-12-01 11:00:05"
+        )
+        self._test_undelivered_pending_responses(
+            subtask_id                         = '8',
+            client_public_key                  = self._get_encoded_provider_public_key(),
+            expected_pending_responses_receive = [
+                PendingResponse.ResponseType.ForceReportComputedTaskResponse,
+            ]
+        )
+        self.assertEqual(len(mail.outbox), len(settings.ADMINS))
```

Comment:
Sorry, I am not sure what is being compared here...

Reply:
If each admin get an email. In issue there is

> An e-mail notification when the last active subtask switches to a passive state would also be useful. But just once so that the admin is not flooded with e-mails when the system stops.